New node shape: diamond

rogalski opened the pull request:
Hello,

I needed a new shape for node available, so I thought I may as well open a pull request.
Unfortunately I think I'll need some help, since my tex knowledge is basically non-existent.

I'll attach relevant info in code comments.

Regards.

rogalski commented on tex/pgfcircshapes.tex:
```diff
@@ -92,6 +92,23 @@
+	\behindforegroundpath{		
+		\pgfscope
+			\pgftransformrotate{45}
+			\pgfpathrectanglecorners{\pgfpoint{\pgf@x+\pgfkeysvalueof{/tikz/circuitikz/nodes width}*\pgfkeysvalueof{/tikz/circuitikz/bipoles/length}*2}{\pgf@y-\pgfkeysvalueof{/tikz/circuitikz/nodes width}*\pgfkeysvalueof{/tikz/circuitikz/bipoles/length}*2}}{\pgfpoint{\pgf@x-\pgfkeysvalueof{/tikz/circuitikz/nodes width}*\pgfkeysvalueof{/tikz/circuitikz/bipoles/length}*2}{\pgf@y+\pgfkeysvalueof{/tikz/circui ...
```
All right, that looks very ugly.
Can `\pgfkeysvalueof{/tikz/circuitikz/nodes width}*\pgfkeysvalueof{/tikz/circuitikz/bipoles/length}*2` be extracted to some kind of temporary variable here?

sistlind replied:
You can use the dimensions \pgf@circ@res@other, \pgf@circ@res@temp and \pgf@circ@res@step as temporary variables. To assign it with a mathematical expression use
\pgfmathsetlength, like e.g. here: https://github.com/circuitikz/circuitikz/blob/master/tex/pgfcircbipoles.tex#L364

rogalski commented on tex/pgfcircbipoles.tex:
```diff
@@ -1718,14 +1718,21 @@
+\ctikzset{d-d/.style =  {\circuitikzbasekey/bipole/nodes/left=diamond, \circuitikzbasekey/bipole/nodes/right=diamond}}
 
 \tikzset{reversed/.style = {\circuitikzbasekey/bipole/reversed=true}}
 
```
I have a huge problem to get around extending \def\drawpoles for another shape. If I understand correctly, code does not actually checks value of `\ctikzvalof{bipole/nodes/left}`, but somehow conditionally declares and modifies `\pgf@circ@temp ` which is a macro?

Why it cannot be done by some simple if/elif/else branching?

rogalski commented on tex/pgfcircbipoles.tex:
```diff
@@ -1718,14 +1718,21 @@
 
 %% Easily usable styles
 
+\ctikzset{o-o/.style =  {\circuitikzbasekey/bipole/nodes/left=empty, \circuitikzbasekey/bipole/nodes/right=empty}}
```
Oh, one space too much after equal sign. I;m sorry about that, I'll fix that in next patch.

rogalski commented:
Oh, and a quick question, is there a reasonable chance for this getting merged to main branch? I don't want to bother you if you will be reluctant to merge it at the end.

rogalski commented:
I just squashed and rebased fixes. Thanks @sistlind for fixing CI environment and for comments!

It looks like it's finished for my requirement, so I won't spend much more time on it. If you want it in main line, I'll be happy to fix any issues discovered in code review.
